src/Compilers/Core/Portable/Text/ChangedText.cs

```diff
@@ -27,11 +27,38 @@ public ChangedText(SourceText oldText, SourceText newText, ImmutableArray<TextCh
             RoslynDebug.Assert(oldText != null);
             Debug.Assert(oldText != newText);
             Debug.Assert(!changeRanges.IsDefault);
+            AssertChangeRangesAreValid(oldText, newText, changeRanges);
 
             _newText = newText;
             _info = new ChangeInfo(changeRanges, new WeakReference<SourceText>(oldText), (oldText as ChangedText)?._info);
         }
 
+        private static void AssertChangeRangesAreValid(
```

Comment:
This uses the verbiage `Assert` which is commonly debug only but this runs in all configurations. Would prefer a different name here like `Requires`

Reply:
WFM.